rules/framework.bzl

```diff
@@ -896,12 +904,14 @@ def _apple_framework_packaging_impl(ctx):
                 avoid_deps.append(dep)
 
     # If we link dynamic - then package it as dynamic
+    plist_out = None
     if ctx.attr.link_dynamic:
         bundle_outs = _bundle_dynamic_framework(ctx, is_extension_safe = is_extension_safe, avoid_deps = avoid_deps)
         avoid_deps_info = AvoidDepsInfo(libraries = depset(avoid_deps + ctx.attr.deps).to_list(), link_dynamic = True)
     else:
         bundle_outs = _bundle_static_framework(ctx, is_extension_safe = is_extension_safe, outputs = outputs)
         avoid_deps_info = AvoidDepsInfo(libraries = depset(avoid_deps).to_list(), link_dynamic = False)
+        plist_out = bundle_outs.providers[0].infoplist
```

Comment:
I don't think we should commit this since it's assuming the order of the returned `providers` stays the same. That seems bad to me. Move to `_get_framework_files` might mitigate this assumption here by relying on more general infrastructure.

Reply:
➕ yeah agree - that was my main question about the PR too - Is it possible to align with the others? I was really hoping to keep 2 distinct branches in this if statement. If it's not possible maybe we can try to improve the code and re-work it as a followup.